docs/architecture.md

```diff
@@ -3,16 +3,64 @@ id: architecture
+
+The application makes use of React Redux as well, this is used as a general state store to keep track of important parameters that are used throughout the application. More on React Redux can be found here [^react-redux].
+
+Most features also contain a file called `middleware.js`. This file acts as a bridge between the component and the functionality of the rest of the application. 
+
+**./modules/external-api**
+In this folder, the external API can be found. This API can be used in various events like participants joining/leaving the meeting, changes in avatars or chat, as well as errors in using the microphone or camera.
+
+**./android and ./ios**
+Both of these folders contain the basics of the Android and iOS app respectively. However, the code for the application itself and its components can be found in the **react/features** folder, which will be explained later in this section. 
+
+**./conference.js**
+This file can be found in the root of the project, and contains the foundation of any interaction between a user and a conference room. This consists of setting up a connection to it, joining the meeting room, muting and unmuting, but also functions to gather information about the participants that are in the room. 
+
+**./lang** 
+This folder contains all the different translations that are present in Jitsi Meet. The translations can be found in the code with each of the keys in the translation maps that can be found in `main-[language].json` files. 
+
+**./css**
+This folder contains all the css that is used in the project. The files (mostly .scss files[^scss]) are split up into features like the React features that they are used in.
+
+## Testing
+The main form of testing code changes is done through torture tests, next to this the code is tested manually. Manual testing is performed while doing code reviews, however there are also testing releases which can be freely downloaded and deployed, or can be used on the [beta test server](https://beta.meet.jit.si/).
+
+An automatic testing pipeline is not built into the Jitsi Meet repository, thus code coverage is not taken into account. However, unit tests can be performed on the Jitsi Meet code base by using the [Jitsi Meet Torture](https://github.com/jitsi/jitsi-meet-torture) project. The project contains automated Maven tests for several key functions such as peer2peer and invites. The testing can be done for IOS, Android and web, which are all the platforms that Jitsi Meet can be used on. The testing is only used in some cases where it is used in combination with the continuous integration by a Jenkins instance running Maven tests. The test results can be viewed on an automatically generated web page.
```

Comment:
This is not correct. We run torture tests, which are end-to-end tests on every pull request. So there is a an automatic testing pipeline setup.

Reply:
Also, not sure if the mention of code coverage adds any value here.

Reply:
> This is not correct. We run torture tests, which are end-to-end tests on every pull request. So there is a an automatic testing pipeline setup.

Hey @saghul, the torture tests are not visible in the "simple CI" which ran over our own PR's, so we assumed it is not run. Could you explain in what way the torture tests are deployed or link some reference to the testing pipeline? We will rephrase this section accordingly.

Reply:
Sure thing. See this PR for example: https://github.com/jitsi/jitsi-meet/pull/8954 --  There is "Simple CI" and there is "default". The latter is Jenkins. It doesn't run for non-project member PRs unless we invoke it by commenting with "Jenkins please test this please". This is for security reasons, since we are running random code on our machines. The pipeline is simple: the PR code is torture-tested against meet.jit.si.

Reply:
Ahh that makes sense, thank you! I will update the text

Reply:
Could you please have a look at the new text?